### PR TITLE
Let's please use our usually used CI/CD pipeline

### PR DESCRIPTION
**Pull request by simar0at**
I got your branch to work with some effort on our usually used CI/CD pipeline.  
https://next-ci-cd.acdh-cluster-2.arz.oeaw.ac.at/
At least for this project I want to use this approach and not have a custom build pipeline or Dockerfile.

### CODE REVIEW

**Comment by stefanprobst** [comment]
this is missing some build args for the docker image (e.g. the redmine service id for the imprint service via `NUXT_PUBLIC_REDMINE_ID`)

also, if generating the preview deployment relies on branch naming conventions (see https://github.com/acdh-oeaw/vicav-vue3/pull/56/commits/41b9f19eebecb417d8a73a7bef1def590fb6dccb) these should be documented somewhere?

may i ask what was not working in the current `build-deploy.yml`?

File: nuxt.config.ts [hunk 1]
```diff
@@ -1,5 +1,13 @@
 import { fileURLToPath } from "node:url";
 
+let nuxtdevtools_installed;
```
Comment (stefanprobst) [inline nuxt.config.ts:3]: why is this necessary?

Comment (simar0at) [reply nuxt.config.ts:3]: Well it isn't but `pnpm prune --prod` does not work without it and that is an integral part of the standardized process.
Try to remove the code and do the following manually:
* `pnpm install`
* `pnpm run build`
* `pnpm prune --prod`

pnpm prune removes about 250 packages form node_modules that don't have much business in the container on the production instance. Among them the nuxt dev tools.

**Comment by simar0at** [comment]
> this is missing some build args for the docker image (e.g. the redmine service id for the imprint service via `NUXT_PUBLIC_REDMINE_ID`)
> 
If you need that (I don't) you can add whatever environment variable is necessary in the GitHub variables as K8S_SECRET_A_VAR with my process. So e.g. K8S_SECRET_NUXT_PUBLIC_REDMINE_ID
> also, if generating the preview deployment relies on branch naming conventions (see [41b9f19](https://github.com/acdh-oeaw/vicav-vue3/commit/41b9f19eebecb417d8a73a7bef1def590fb6dccb)) these should be documented somewhere?
> 
This is some of the first projects that use that so a documentation is missing. A part of that documentation is probably just the code in starter.yaml that creates that.
> may i ask what was not working in the current `build-deploy.yml`?

I don't have time to maintain it and it is invented in house and not used by many more institutions on the internet.  
The standard workflow is used by a lot of people at heroku, some more that bought into gitlab AutoDevOps and I think some more. Many more eyes probably catch more errors.

File: components/app-header.vue [hunk 2]
```diff
@@ -105,7 +105,7 @@ function onSelectMenuItem(item: ItemType) {
 				kind: "text",
 				params: {
 					// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-					id: item.id!,
+					id: item.target!,
```
Comment (stefanprobst) [inline components/app-header.vue:108]:
```suggestion
					id: item.id!,
```

**Comment by stefanprobst** [comment]
> I don't have time to maintain it and it is invented in house

i mean all it does is clean up the `starter.yaml` config and use a pretty standard Dockerfile to build the docker image

File: components/app-header.vue (hunk 2 above)
Comment (stefanprobst) [reply components/app-header.vue:108]: endpoint actually accepts `target`, not `id`
